src/cbc_sdk/platform/alerts.py

```diff
@@ -997,6 +997,92 @@ def set_rows(self, rows):
+
+        Note:
+            - `range` will take precendent over `start` and `end` if provided.
+
+        Examples:
+            >>> query = api.select(Event).set_time_range(start="2020-10-20T20:34:07Z")
+            >>> second_query = api.select(Event).
+            ...     set_time_range(start="2020-10-20T20:34:07Z", end="2020-10-30T20:34:07Z")
+            >>> third_query = api.select(Event).set_time_range(range='-3d')
+        """
+        args_count = args.__len__()
+        start = kwargs.get("start")
+        end = kwargs.get("end")
+        range = kwargs.get("range")
+
+        if args_count > 0:
+            key = args[0]
+            if key not in ["create_time", "first_event_time", "last_event_time", "last_update_time",
+                           "backend_timestamp", "backend_update_timestamp"]:
+                raise ApiError("key must be one of create_time, first_event_time, last_event_time, backend_timestamp,"
+                               " backend_update_timestamp, or last_update_time")
+            else:
+                return self.set_time_criterion_filter(*args, **kwargs)
+        else:
+            # everything before this is only for backwards compatibuility, once v6 deprecates all thes other
+            # checks can be removed
+            self._time_range = {}
+            if start:
+                if not isinstance(start, str):
+                    raise ApiError(f"Start time must be a string in ISO 8601 format. {start} is a {type(start)}.")
+                self._time_range["start"] = start
+            if end:
+                if not isinstance(end, str):
+                    raise ApiError(f"End time must be a string in ISO 8601 format. {end} is a {type(end)}.")
+                self._time_range["end"] = end
+            if range:
+                if not isinstance(range, str):
+                    raise ApiError(f"Range must be a string. {range} is a {type(range)}.")
+                self._time_range["range"] = range
+            return self
+
+    def set_time_criterion_filter(self, key, **kwargs):
+        """
+        Restricts the alerts that this query is performed on to the specified time range.
+
+        The time may either be specified as a start and end point or as a range.
+
+        Args:
+            key (str): The key to use for criteria one of create_time,
+                       first_event_time, last_event_time, or last_update_time
+            **kwargs (dict): Used to specify start= for start time, end= for end time, and range= for range.
+
+        Returns:
+            AlertSearchQuery: This instance.
+        """
+        if key not in ["create_time", "first_event_time", "last_event_time", "last_update_time", "backend_timestamp",
+                       "backend_update_timestamp"]:
+            raise ApiError("key must be one of create_time, first_event_time, last_event_time, backend_timestamp,"
```

Comment:
Valid fields for v7 are:
backend_timestamp
backend_update_timestamp
detection_timestamp
first_event_timestamp
last_event_timestamp
mdr_determination_change_timestamp
mdr_workflow_change_timestamp
user_update_timestamp
workflow_change_timestamp